src/bmp280.c

```diff
@@ -24,47 +24,45 @@ static uint16_t                 bmp_cs_pin = 0;
 bmp280_calib_digit_s            calib;
 
 
-ErrorStatus bmp280_init(SPI_TypeDef *SPI_nr, GPIO_TypeDef* CS_GPIO, uint16_t CS_GPIO_Pin)
-{
-    ErrorStatus ret = SUCCESS;
-    
-    ret |= spi_init(&bmp_spi_dev, SPI_nr);
+hub_retcode_t bmp280_init(SPI_TypeDef *SPI_nr, GPIO_TypeDef* CS_GPIO, uint16_t CS_GPIO_Pin)
+{    
+    if (OK != spi_init(&bmp_spi_dev, SPI_nr)) return INIT_ERROR;
 
-    ret |= spi_cs_init(CS_GPIO, CS_GPIO_Pin);
+    spi_cs_init(CS_GPIO, CS_GPIO_Pin);
     bmp_cs_gpio = CS_GPIO;
     bmp_cs_pin = CS_GPIO_Pin;
 
-    bmp280_get_calib();
-
-    return ret;
+    return bmp280_get_calib();
 }
 
 
-ErrorStatus bmp280_get_calib(void)
+hub_retcode_t bmp280_get_calib(void)
 {
-    ErrorStatus ret = SUCCESS;
+    hub_retcode_t ret = OK;
 
     // get calibration data
     uint8_t dig[26];
     BMP280_READ_M(BMP280_dig_T1, dig, BMP280_dig_T_size + BMP280_dig_P_size);
```

Comment:
Gdzieś już pisałem, że to makro to antypattern, ale no jak chcesz

Reply:
że zamiast OK ma być 0 ?